Added documentation for property precedence order, per-tenant configuration and /v1/health/config endpoint.

vnandwana opened the pull request:
Ticket -- https://github.com/killbill/technical-support/issues/216

vnandwana committed:
Added documentation for property precedence order, per-tenant configuration and /v1/health/config endpoint.

vnandwana commented on userguide/aviate/aviate-health.adoc:
```diff
@@ -27,6 +27,74 @@ com.killbill.billing.plugin.aviate.enableHealthReporter=true
+
+Includes the `user.timezone` property, which is force set to `GMT`.
+
+This property cannot be overridden.
```
Overriding `user.timezone` results in the following warning:

`Overwrite of user.timezone system property with IST may break database serialization of date. Kill Bill will overwrite to GMT`

[This code](https://github.com/killbill/killbill-platform/blob/master/base/src/main/java/org/killbill/billing/platform/config/DefaultKillbillConfigSource.java#L213) resets the value to GMT.

reshmabidikar reviewed (commented):
I do not quite follow how this documentation fits into Aviate Health. Does this correspond to a new API endpoint that exposes all the config properties? Is this endpoint provided by the aviate plugin? If so, a couple of suggestions:

- Add the documentation for this endpoint in the API docs [here](https://apidocs.killbill.io/aviate-health#health-apis)
- Add a new section (maybe below the [Metrics](https://docs.killbill.io/latest/aviate-health#_aviate_metrics) section) to the [Aviate health doc](https://docs.killbill.io/latest/aviate-health) and move this content to this section. Ensure that the [API doc](https://apidocs.killbill.io/aviate-health#health-apis) points to this documentation.

sbrossie commented:
@vnandwana Will you rebase this PR based on latest `v3` and resubmit any changes that you think are relevant?

vnandwana committed:
Rebased to the latest v3 branch and made required changes to the Runtime Properties section.

vnandwana commented:
> I do not quite follow how this documentation fits into Aviate Health. Does this correspond to a new API endpoint that exposes all the config properties? Is this endpoint provided by the aviate plugin? If so, a couple of suggestions:
> 
> * Add the documentation for this endpoint in the API docs [here](https://apidocs.killbill.io/aviate-health#health-apis)
> * Add a new section (maybe below the [Metrics](https://docs.killbill.io/latest/aviate-health#_aviate_metrics) section) to the [Aviate health doc](https://docs.killbill.io/latest/aviate-health) and move this content to this section. Ensure that the [API doc](https://apidocs.killbill.io/aviate-health#health-apis) points to this documentation.

Hi @reshmabidikar, thank you for your suggestions. I have addressed these comments, please review the changes at your convenience. This branch has been rebased onto the latest v3.


API doc PR -- https://github.com/killbill/slate/pull/352

reshmabidikar commented on userguide/aviate/aviate-health.adoc:
```diff
@@ -97,16 +97,69 @@ The metrics exposed by the Aviate plugin can mainly be categorized in the follow
+
+This provides fine-grained control in multi-tenant deployments, enabling different tenants to customize selected settings while still inheriting defaults from the global configuration.
+
+For more details, see the official documentation: https://docs.killbill.io/latest/plugin_installation#per-tenant-configuration.
```
I think it would be better to point to [this link](https://docs.killbill.io/latest/userguide_configuration#_per_tenant_properties) as  https://docs.killbill.io/latest/plugin_installation#per-tenant-configuration has more to do with per-tenant plugin configuration.

vnandwana replied:
Plz share more info on this, the present link is the same as the suggested link.

reshmabidikar replied:
I'm suggesting https://docs.killbill.io/latest/userguide_configuration#_per_tenant_properties, the present link is https://docs.killbill.io/latest/plugin_installation#per-tenant-configuration..

reshmabidikar commented on userguide/aviate/aviate-health.adoc:
```diff
@@ -97,16 +97,69 @@ The metrics exposed by the Aviate plugin can mainly be categorized in the follow
+using value from 'EnvironmentVariables': '11s'`
+
+
+To make it easier to understand the runtime values of these properties, Kill Bill provides the `/v1/health/config` endpoint.
```
Which endpoint does this refer to? Can we point to the doc link for this endpoint?

vnandwana replied:
PR for this endpoint is https://github.com/killbill/slate/pull/352 -- if we need to link this API doc, we'll need to merge the Slate PR.

reshmabidikar commented on userguide/aviate/aviate-health.adoc:
```diff
@@ -97,16 +97,69 @@ The metrics exposed by the Aviate plugin can mainly be categorized in the follow
+
+* Runtime Configuration
+
+Properties loaded from a configuration file specified via `-Dorg.killbill.server.properties=<propertyFile>`.
```
Does this also include the properties defined in `catalina.properties` in case of a Tomcat configuration? If so, would be good to mention this explicitly.

vnandwana replied:
No, it doesn't.

reshmabidikar reviewed (commented):
Looks good, I've left a few minor comments.

sbrossie reviewed (commented):
Given we already merged https://github.com/killbill/killbill-docs/pull/627, was this branch rebased to include it?

vnandwana commented:
> Given we already merged #627, was this branch rebased to include it?

Yes, this branch is rebased to the latest v3 branch -- here is the commit for reference [26e6981](https://github.com/killbill/killbill-docs/pull/632/commits/26e6981f86d8c2a9ad37ad297a8a9746ce29cb8a)

vnandwana committed:
Added API endpoint for /health/config endpoint.
